### ISSUE TITLE
float32 vector_matrix action loses accuracy for small rotations

### ISSUE THREAD

**Issue by Concode0** [Concode0/Versor#30]
### What happened?

On the `next` branch, the `vector_matrix` action route can exceed clifra's fixed float32 correctness tolerance even for a minimal Euclidean 2-D rotation.

The same semantic action through `rotor_product` passes, and the corresponding float64 `vector_matrix` result is accurate.

Current investigation suggests that the discrepancy may originate in `torch.matrix_exp` rather than clifra's algebra/layout logic, but this has not yet been established conclusively. This issue tracks the clifra-side symptom until the upstream diagnosis is complete.

### Where?

`clifra.core._kernel.execution.action`, `vector_matrix` route for bivector-generated actions.

### Algebra signature

Related float32 discrepancies also appear in higher-dimensional homogeneous action cases.

### Minimal reproduction

```python
import torch
from clifra import make_algebra

algebra = make_algebra(2, 0, dtype=torch.float32)
vectors = algebra.layout((1,))
bivectors = algebra.layout((2,))

action = algebra.plan_bivector_action(
    input=vectors,
    generator=bivectors,
    output=vectors,
)

x = torch.tensor([0.5, 1.0], dtype=torch.float32)
b = torch.tensor([0.5], dtype=torch.float32)

print(action(x, b))
```

### Error output

```shell
float32 CPU, default/vector_matrix:
  actual   = [-0.0406067371, 1.1172881126]
  expected = [-0.0406342577, 1.1172953312]
  max abs error ≈ 2.752e-5

float32 CPU, rotor_product:
  max abs error ≈ 6.60e-8

float64 CPU, vector_matrix:
  max abs error ≈ 5.55e-17
```

### Environment

Branch: next, PyTorch: 2.10.0, Tested on: CPU float32 CPU float64 MPS float32

**Comment by Concode0** [comment]
Check ea81e11947cba62fc3a99f8b6366cadb0ad815fa; planning to keep it until the issue is fixed upstream.

**Comment by Concode0** [comment]
Upstream diagnosis is now complete. The discrepancy was caused by an incorrect coefficient in PyTorch's T8 matrix_exp approximation, not by clifra's algebra/layout logic.

Upstream issue: pytorch/pytorch#196592
Upstream fix: pytorch/pytorch#196658

clifra currently keeps a singleton matrix_exp workaround for affected PyTorch releases. Keeping this issue open until the fix is available in a released PyTorch version and the compatibility/workaround policy is finalized.
